plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -178,7 +183,7 @@ public List<HostAddress> getAddresses()
 
     public List<String> getAllSchemas()
     {
-        return ImmutableList.copyOf(client.listDatabaseNames()).stream()
+        return ImmutableList.copyOf(listDatabaseNames()).stream()
```

Comment:
please look below  - change the return type of listDatabaseNames to make it simpler to read. I'm kind of lost on what is transformed into what.